blockchain/blockchain.go

```diff
@@ -622,6 +651,21 @@ func stateUpdateByHash(txn db.Transaction, hash *felt.Felt) (*core.StateUpdate,
 	})
 }
 
+func l1HandlerTxnHashByMsgHash(txn db.Transaction, l1HandlerMsgHash *common.Hash) (*felt.Felt, error) {
+	var raw []byte
+	err := txn.Get(db.L1HandlerTxnHashByMsgHash.Key(l1HandlerMsgHash.Bytes()), func(val []byte) error {
+		if len(val) == 0 {
+			return db.ErrKeyNotFound
+		}
+		raw = val
```

Comment:
This is just a shallow copy which can have unintended consequences, you should do a deep copy. If you notice whenever we fetch data from DB almost always there is an unmarshal function that serves the same purpose as a deep copy.

Reply:
You can use the unmarshal function on felt added in PR: #2246

Reply:
Thanks for catching this and the suggestion, I've used the above PR to make a hard-copy.